django/db/models/deletion.py

```diff
@@ -51,6 +52,23 @@ def DO_NOTHING(collector, field, sub_objs, using):
     pass
 
 
+def get_candidate_relations_to_delete(opts):
+    # Collect models that contain candidate relations to delete. This may include
+    # relations coming from proxy models.
+    candidate_models = {opts}
+    candidate_models = candidate_models.union(opts.concrete_model._meta.proxied_children)
+    # For each model, get all candidate fields.
+    candidate_model_fields = chain.from_iterable(
+        opts.get_fields(include_hidden=True) for opts in candidate_models
+    )
+    # Keep related objects not coming from ManyToManyFields from all candidate
+    # model fields.
+    return (
+        f for f in candidate_model_fields
+        if f.auto_created and not f.concrete and not f.many_to_many
```

Comment:
For me, something tied to `auto_created` makes more sense - what we're trying to capture here is whether the field is a canonical representation, or the "other" side that needs to exist for data access purposes, but doesn't require explicit action (because the action will happen on the canonical side). 

If we were to tie this to `one_to_one`/`one_to_many`, then we're assuming that a `one_to_one`/`one_to_many` field will always be the canonical representation requiring action. GenericRelations show how this isn't necessarily the case. 

A theoretical "OneToManyField" (i.e., a "reverse" definition of a ForeignKey) would be another interesting case - that's a field where the "concrete" column is on the reverse, auto-created object.